runtime/commonMain/src/kotlinx/serialization/modules/SerialModule.kt

```diff
@@ -1,60 +1,140 @@
 /*
- * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 
-@file:Suppress("UNCHECKED_CAST", "RedundantVisibilityModifier")
-
 package kotlinx.serialization.modules
 
 import kotlinx.serialization.*
+import kotlinx.serialization.internal.*
 import kotlin.reflect.*
 
 /**
  * [SerialModule] is a collection of serializers used by [ContextSerializer] and [PolymorphicSerializer]
  * to override or provide serializers at the runtime, whereas at the compile-time they provided by the serialization plugin.
+ * It can be considered as a map where serializers can be found using their statically known KClasses.
  *
- * It can be considered as a map where serializers are found using statically known KClasses.
- *
- * To enable runtime resolution of serializers, one of the special annotations must be used on target types
- * and a serial module with serializers should be used during construction of [SerialFormat].
+ * To enable runtime serializers resolution, one of the special annotations must be used on target types
+ * ([Polymorphic] or [ContextualSerialization]), and a serial module with serializers should be used during construction of [SerialFormat].
  *
  * @see ContextualSerialization
  * @see Polymorphic
  */
-public interface SerialModule {
+public sealed class SerialModule {
 
     /**
      * Returns a contextual serializer associated with a given [kclass].
      * This method is used in context-sensitive operations on a property marked with [ContextualSerialization] by a [ContextSerializer]
      */
-    public fun <T : Any> getContextual(kclass: KClass<T>): KSerializer<T>?
+    public abstract fun <T : Any> getContextual(kclass: KClass<T>): KSerializer<T>?
 
     /**
      * Returns a polymorphic serializer registered for a class of the given [value] in the scope of [baseClass].
      */
-    public fun <T : Any> getPolymorphic(baseClass: KClass<T>, value: T): KSerializer<out T>?
+    public abstract fun <T : Any> getPolymorphic(baseClass: KClass<T>, value: T): KSerializer<out T>?
 
     /**
      * Returns a polymorphic serializer registered for for a [serializedClassName] in the scope of [baseClass].
      */
-    public fun <T : Any> getPolymorphic(baseClass: KClass<T>, serializedClassName: String): KSerializer<out T>?
+    public abstract fun <T : Any> getPolymorphic(baseClass: KClass<T>, serializedClassName: String): KSerializer<out T>?
+
+    /**
+     * Returns a default polymorphic serializer registered for a [serializedClassName] in the scope of [baseClass].
+     * In contrast with [getPolymorphic], module user registers a factory function for default serializers that can
+     * provide a serializer even for an unknown in advance [serializedClassName].
+     */
+    public abstract fun <T : Any> getDefaultPolymorphic(
```

Comment:
Why do we need an extra method instead of regular `getPolymorphic(baseClass, serializedClassName)`?